src/lucky_record/errors.cr

```diff
@@ -0,0 +1,14 @@
+module LuckyRecord
+  # = Lucky Record Errors
+  #
+  # Generic Lucky Record exception class.
+  class LuckyRecordError < Exception
+  end
+
+  # Raised when Lucky Record cannot find a record by given id
+  class RecordNotFoundError < LuckyRecordError
+    def initialize(@model : Symbol, @id : Int32)
```

Comment:
One last thing. Could you make the id also accept a string. Right now it’s possible to use a String because LuckyRecord casts it to an Int32. 

You can add a test for this by duplicating what you have but passing “id” in.

Reply:
If i pass "id" I get <PQ::PQError: invalid input syntax for integer: "id">, only if I pass "1" RecordNotFoundError will be raised, so should I make the test with "1"?

Reply:
Hmm that's a good question. I'm going to need to pull this down and take a look I think. If I don't get back to you in a couple days can you ping me to remind me to take a look?

Reply:
Sure, no problem, I think we need to try to cast before and raise the correct error before PG does.

Reply:
It should cast when you call `id`. I think for now what we can do is accept a string in the Error and call `...Error.new(id: id.to_s)` to make sure it is a String when it gets to the error. In the long term I'm going to make it so that there is more type restriction, but until that's done this should do the trick.

```crystal
... || LuckyRecord::RecordNotFoundError.new(model: @@table_name, id: id.to_s)
```